nexus/types/src/external_api/params.rs

```diff
@@ -1117,11 +1131,11 @@ pub struct ImportBlocksBulkWrite {
 /// Parameters for finalizing a disk
 #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
 pub struct FinalizeDisk {
-    #[serde(flatten)]
-    pub project_selector: Option<ProjectSelector>,
-
-    /// an optional snapshot name
-    pub snapshot_name: Option<String>,
+    /// If specified a snapshot of the disk will be created with the given
+    /// name at the end of finalization. If not specified, a snapshot for
+    /// the disk will _not_ be created. A snapshot can be manually created
+    /// once the disk transitions into the `Detached` state.
+    pub snapshot_name: Option<Name>,
```

Comment:
@jmpesp could you read over this and let me know if I've represented it accurately?

Reply:
I would reword it from `at the end of finalization` to `during finalization`, but otherwise this is accurate.

It also makes **way** more sense for this to be Name instead of String! Thanks for catching that.